lib/iris/unit.py

```diff
@@ -1875,6 +1876,11 @@ def convert(self, value, other, ctype=FLOAT64):
                         if issubclass(value_copy.dtype.type, np.integer):
                             value_copy = value_copy.astype(
                                 _ctypes2numpy[ctype])
+                        # Convert big-endian arrays because udunits doesn't
```

Comment:
We don't know that this is true for big-endian systems. Alas, I don't propose we check that - the tests will catch the problem if it comes up :+1: